### PR TITLE
xdsclient: populate error details for NACK

### LINKED ISSUE #3969: xds: NACKs missing error detail
### What version of gRPC are you using?
1.33.1

### What version of Go are you using (`go version`)?
1.14

### What operating system (Linux, Windows, …) and version?
Linux

### What did you do?
Sent an invalid XDS response

### What did you expect to see?
NACK with ErrorDetails set. From the spec "If Envoy had instead rejected configuration update X, it would reply with error_detail populated"

### What did you see instead?
ErrorDetails unset

### PR DESCRIPTION
**Pull request by menghanl**
fixes #3969

### CODE REVIEW

File: xds/internal/client/transport_helper.go [hunk 1]
```diff
@@ -246,16 +246,16 @@ func (t *TransportHelper) send(ctx context.Context) {
...
+				rType                  ResourceType
+				version, nonce, errMsg string
+				send                   bool
 			)
 			switch update := u.(type) {
 			case *watchAction:
 				target, rType, version, nonce = t.processWatchInfo(update)
 			case *ackAction:
-				target, rType, version, nonce, send = t.processAckInfo(update, stream)
+				target, rType, version, nonce, errMsg, send = t.processAckInfo(update, stream)
```
Comment (easwars) [inline xds/internal/client/transport_helper.go:258]: Why does `processAckInfo` have to return the `errMsg`. Why can't we use it directly here? It is available in `update`.

**Review by howardjohn — APPROVED** [review]

File: xds/internal/client/transport_helper.go (hunk 1 above)
Comment (menghanl) [reply xds/internal/client/transport_helper.go]: Right. This is unnecessary, and `processAckInfo` cannot change it anyway. Done.

File: xds/internal/client/transport_helper.go [hunk 2]
```diff
@@ -246,10 +246,10 @@ func (t *TransportHelper) send(ctx context.Context) {
 			t.sendCh.Load()
 
 			var (
-				target         []string
-				rType          ResourceType
-				version, nonce string
-				send           bool
+				target                 []string
+				rType                  ResourceType
+				version, nonce, errMsg string
```
Comment (easwars) [inline xds/internal/client/transport_helper.go:251]: Nit: We might as well get rid of `errMsg` and use `update.errMsg` directly.

**Review by easwars — APPROVED** [review]

File: xds/internal/client/transport_helper.go (hunk 2 above)
Comment (menghanl) [reply xds/internal/client/transport_helper.go:251]: There's a `switch: case`. Only `ackAction` has the `errMsg` field.

**Pull request merged**